serverless/resource_serverless.go

```diff
@@ -0,0 +1,158 @@
+package serverless
+
+import (
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceServerless() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceServerlessCreate,
+		Read:   resourceServerlessRead,
+		Update: resourceServerlessUpdate,
+		Delete: resourceServerlessDelete,
+
+		Schema: map[string]*schema.Schema{
+			"config_dir": &schema.Schema{
```

Comment:
@tptee -- Can you remind me why this is required? I would assume it should default to CWD and not be required.

Reply:
@tptee -- Bump ^^^ . Is `config_dir` really required or should we default it?

Reply:
Filed https://github.com/FormidableLabs/terraform-provider-serverless/issues/11 if we're not taking this issue up in this PR.

Reply:
I haven't run into a project where this default makes sense–the `serverless.yml` is usually colocated with the application code, not the Terraform module...

Reply:
Maybe I'm not understanding here, "current working directory" is almost always where `serverless.yml` is located and we also usually run TF commands out of CWD?

Reply:
Actually, I got you. We usually `cd terraform/` before commands, and most folks don't have `main.tf` where `serverless.yml` is.

Reply:
In projects I've worked on, we're only are able to run our Terraform in `cwd` because we add wrapper scripts around it (e.g. `--terragrunt-working-dir` or `cd terraform/some-module &&`. I haven't seen a Terraform module and a `serverless.yml` be in the same directory on disk yet, although it's possible!

Reply:
oh right, what you said :D